app/models/game.rb

```diff
@@ -1,45 +1,54 @@
 # model Game,piece creation inside
+# DO NOT chnage the sequence of methods
+# or you will break the front end view
 class Game < ActiveRecord::Base
   has_many :pieces
   has_many :players
 
-  after_create :populate_bishops!, :populate_rooks!, :populate_pawns!,\
-               :populate_knights!, :populate_queens_kings!
+  after_create :populate_left_black_half!,\
+               :populate_right_black_half!,\
+               :populate_black_pawns!,\
+               :populate_white_pawns!,\
+               :populate_left_white_half!,\
+               :populate_right_white_half!
```

Comment:
Hmmm I'm not sure why I didn't comment on this initially, but why are we using all these methods? Wouldn't it be simpler to have one method for populate_board?